avm/res/resources/deployment-script/main.bicep

```diff
@@ -0,0 +1,275 @@
+  name: '46d3xbcp.res.resources-deploymentscript.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+// ================ //
+// Resources        //
+// ================ //
+
+resource deploymentScript 'Microsoft.Resources/deploymentScripts@2023-08-01' = {
+  name: name
+  location: location
+  tags: tags
+  identity: identity
+  kind: any(kind)
+  properties: {
+    azPowerShellVersion: kind == 'AzurePowerShell' ? azPowerShellVersion : null
+    azCliVersion: kind == 'AzureCLI' ? azCliVersion : null
+    containerSettings: !empty(containerGroupName) ? containerSettings : null
+    storageAccountSettings: !empty(storageAccountResourceId) ? storageAccountSettings : null
+    arguments: arguments
+    environmentVariables: !empty(environmentVariables) ? environmentVariables.secureList : []
+    scriptContent: !empty(scriptContent) ? scriptContent : null
+    primaryScriptUri: !empty(primaryScriptUri) ? primaryScriptUri : null
+    supportingScriptUris: !empty(supportingScriptUris) ? supportingScriptUris : null
+    cleanupPreference: cleanupPreference
+    forceUpdateTag: runOnce ? resourceGroup().name : baseTime
+    retentionInterval: retentionInterval
+    timeout: timeout
+  }
+}
+
+// ================ //
+// Outputs          //
+// ================ //
+
+@description('The resource ID of the deployment script.')
+output resourceId string = deploymentScript.id
+
+@description('The resource group the deployment script was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The name of the deployment script.')
+output name string = deploymentScript.name
+
+@description('The location the resource was deployed into.')
+output location string = deploymentScript.location
+
+@description('The output of the deployment script.')
+output outputs object = contains(deploymentScript.properties, 'outputs') ? deploymentScript.properties.outputs : {}
+
+// ================ //
+// Definitions      //
+// ================ //
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type managedIdentitiesType = {
+  @description('Optional. The resource ID(s) to assign to the resource.')
+  userAssignedResourcesIds: string[]
+}?
+
+type roleAssignmentType = {
+  @description('Required. The name of the role to assign. If it cannot be found you can specify the role definition ID instead.')
```

Comment:
This will also need updating to the latest version in the interface [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#role-assignments)